### PR TITLE
refactor: expose core metric names for doc generation 

### PR DESCRIPTION
**Pull request by jeqo**
Add a metric registry for Caching and Thread Pool metrics, document metrics, and expose metric name templates to be used on documentation generation.

### CODE REVIEW

File: core/src/main/java/io/aiven/kafka/tieredstorage/fetch/cache/ChunkCache.java
Comment (jeqo) [inline core/src/main/java/io/aiven/kafka/tieredstorage/fetch/cache/ChunkCache.java:1]: to be used by Metric documentation in the docs module (coming on following PR)

File: core/src/main/java/io/aiven/kafka/tieredstorage/fetch/index/MemorySegmentIndexesCache.java
Comment (jeqo) [inline core/src/main/java/io/aiven/kafka/tieredstorage/fetch/index/MemorySegmentIndexesCache.java:1]: Same here

File: core/src/main/java/io/aiven/kafka/tieredstorage/fetch/manifest/MemorySegmentManifestCache.java
Comment (jeqo) [inline core/src/main/java/io/aiven/kafka/tieredstorage/fetch/manifest/MemorySegmentManifestCache.java:1]: Same here

File: core/src/main/java/io/aiven/kafka/tieredstorage/metrics/CaffeineMetricsRegistry.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.metrics;
+
+import java.util.List;
+
+import org.apache.kafka.common.MetricNameTemplate;
+
+public class CaffeineMetricsRegistry {
```
Comment (jeqo) [inline core/src/main/java/io/aiven/kafka/tieredstorage/metrics/CaffeineMetricsRegistry.java:23]: Similar to MetricsRegistry to expose metric names templates

File: core/src/main/java/io/aiven/kafka/tieredstorage/metrics/MetricsRegistry.java
Comment (jeqo) [inline core/src/main/java/io/aiven/kafka/tieredstorage/metrics/MetricsRegistry.java:1]: Largest change, mostly docs and sorting out

File: core/src/main/java/io/aiven/kafka/tieredstorage/metrics/ThreadPoolMonitorMetricsRegistry.java
Comment (jeqo) [inline core/src/main/java/io/aiven/kafka/tieredstorage/metrics/ThreadPoolMonitorMetricsRegistry.java:1]: Another metric registry for thread pool monitoring

**Review by biggusdonzus — APPROVED** [review]
looks good :+1:

**Pull request merged**